libsql-server/src/replication/snapshot.rs

```diff
@@ -201,17 +276,34 @@ impl SnapshotMerger {
         log_id: Uuid,
     ) -> anyhow::Result<()> {
         let mut snapshots = Self::init_snapshot_info_list(db_path).await?;
-        while let Some((name, size, db_page_count)) = receiver.recv().await {
-            snapshots.push((name, size));
-            if Self::should_compact(&snapshots, db_page_count) {
-                let compacted_snapshot_info =
-                    Self::merge_snapshots(&snapshots, db_path, log_id).await?;
-                snapshots.clear();
-                snapshots.push(compacted_snapshot_info);
+        let mut working = false;
+        let mut job: Pin<Box<dyn Future<Output = anyhow::Result<_>> + Sync + Send>> =
```

Comment:
We can use this type https://docs.rs/tokio-util/latest/tokio_util/sync/struct.ReusableBoxFuture.html

Reply:
nice!